pages/profile/index.tsx

```diff
@@ -0,0 +1,283 @@
+import LanguageRoundedIcon from '@mui/icons-material/LanguageRounded';
+import { useRouter } from 'next/router';
+import Image from 'next/image';
+import React, { useEffect, useRef, useState } from 'react';
+import { useAuthContext } from '../../lib/user/AuthContext';
+import LoadIcon from '../../components/LoadIcon';
+import { getFileExtension } from '../../lib/util';
+import QRCode from '../../components/dashboardComponents/QRCode';
+import GitHubIcon from '@mui/icons-material/GitHub';
+import LinkedInImage from '@/public/icons/linkedin.png';
+import ChickenImage from '@/public/assets/profile-chicken-egg.png';
+import { TextField, TextFieldProps } from '@mui/material';
+import Link from 'next/link';
+import { RequestHelper } from '@/lib/request-helper';
+
+/**
+ * A page that allows a user to modify app or profile settings and see their data.
+ *
+ * Route: /profile
+ */
+export default function ProfilePage() {
+  const router = useRouter();
+  const { isSignedIn, hasProfile, user, profile, updateProfile } = useAuthContext();
+  const [uploading, setUploading] = useState<boolean>(false);
+  const resumeRef = useRef(null);
+
+  useEffect(() => {
+    if (hasProfile && !profile) window.location.reload();
```

Comment:
Any reason why we do this?

Reply:
after we update the profile, if we just do the router.push('/profile') it returns me undefined for profile, i think profile is not being fetched for some reason but the reload works.

Reply:
https://github.com/user-attachments/assets/fa2597e7-6528-4c31-938f-c9cf95b63368